include/ctranslate2/encoder.h

```diff
@@ -13,11 +13,20 @@ namespace ctranslate2 {
     std::future<EncoderForwardOutput>
     forward_batch_async(std::vector<std::vector<std::string>> tokens);
 
+    std::future<EncoderForwardOutput>
+    forward_batch_async(std::vector<std::vector<std::string>> tokens, std::vector<std::vector<size_t>> token_type_ids);
+
     std::future<EncoderForwardOutput>
     forward_batch_async(std::vector<std::vector<size_t>> ids);
 
+    std::future<EncoderForwardOutput>
+    forward_batch_async(std::vector<std::vector<size_t>> ids, std::vector<std::vector<size_t>> token_type_ids);
+
     std::future<EncoderForwardOutput>
     forward_batch_async(const StorageView& ids, const StorageView& lengths);
+    
+    std::future<EncoderForwardOutput>
+    forward_batch_async(const StorageView& ids, const StorageView& lengths, std::vector<std::vector<size_t>> token_type_ids);
```

Comment:
I think we can avoid creating new method overloads by adding the new argument with a default value to the existing methods. Something like:

```cpp
std::future<EncoderForwardOutput>
forward_batch_async(std::vector<std::vector<std::string>> tokens,
                    std::vector<std::vector<size_t>> token_type_ids = {});
```

What do you think?